bson/encoder_example_test.go

```diff
@@ -121,3 +122,126 @@ func ExampleEncoder_UseJSONStructTags() {
 	fmt.Println(bson.Raw(buf.Bytes()).String())
 	// Output: {"name": "Cereal Rounds","sku": "AB12345","price_cents": {"$numberLong":"399"}}
 }
+
+func ExampleEncoder_multipleBSONDocuments() {
+	// Create an Encoder that writes BSON values to a bytes.Buffer.
+	buf := new(bytes.Buffer)
+	vw, err := bsonrw.NewBSONValueWriter(buf)
+	if err != nil {
+		panic(err)
+	}
+	encoder, err := bson.NewEncoder(vw)
+	if err != nil {
+		panic(err)
+	}
+
+	type Coordinate struct {
+		X int
+		Y int
+	}
+
+	// Use the encoder to marshal 5 Coordinate values as a sequence of BSON
+	// documents.
+	for i := 0; i < 5; i++ {
+		err := encoder.Encode(Coordinate{
+			X: i,
+			Y: i,
```

Comment:
Could we give these fields different values, imposing a stricter test example?

```suggestion
			Y: i+1,
```